docs/proposals/scheduling/replicas-assign/divide-replicas-by-static-weight-evenly.md

```diff
@@ -0,0 +1,220 @@
+which is not expected
+
+We expect that it will be assigned to member2 and member3 with equal probability.
+
+
+## Proposal
+
+Since the unevenness of the current implementation is caused by sorting the clusters in dictionary order, 
+the simplest way is changing it to directly randomize the ordering when the weights are equal.
+
+However, it requires consideration of scheduling inertia, avoiding biased rescheduling results due to the introduction of randomization.
+
+Therefore, to solve the problem of uneven replicas allocation, I propose to optimize the ordering of clusters when allocating remainders as:
+
+1. Sort the clusters first by weights
+2. If the weights are equal, then the clusters are sorted according to the current number of replicas
+   (more current number of replicas implies that the remainders of the last scheduling were randomized to such clusters,
+   and in order to keep the inertia in this scheduling, such clusters should also be prioritized).
+3. If the weights are equal and the current number of replicas is equal too, then the clusters will be randomized.
+
+
+## Examples
+
+### E.g.1 Expanded replicas scenario
+
+In expanding replicas scenario, we expect directly increasing replicas in several clusters,
+avoiding different results calculated when rescheduling due to randomness, which leads to some clusters increasing replicas, while others reducing replica.
+
+Yet this proposal can avoid this problem:
+
+```
+sum replicas = 7，clusters = [member1, member2, member3, member4]，weight = 2 : 1 : 1 : 1
+
+1、Calculation：【each cluster allocations】= 2、1、1、1，【remainders】= 2          // 7*2/5=2、7*1/5=1
+2、Sorting：member1 > member2 = member3 = member4
+3、One possible result：3、2、1、1
+
+Supposing sum replicas change from 7 to 8
+
+1、Calculation：【each cluster allocations】= 3、1、1、1，【remainders】= 2          // 8*2/5=3、8*1/5=1
+2、Sorting：member1 > member2 > member3 = member4                                 // member2 has more current replicas than member3 / member4
+3、Result：4、2、1、1  (results ensured inertia)
+```
+
+### E.g.2 Reduced replicas scenario
+
+In reducing replicas scenario, we expect directly reducing replicas in several clusters,
+avoiding different results calculated when rescheduling due to randomness, which leads to some clusters increasing replicas, while others reducing replica.
+
+Yet this proposal can avoid this problem:
+
+```
+sum replicas = 9，clusters = [member1, member2, member3, member4]，weight = 2 : 1 : 1 : 1
+
+1、Calculation：【each cluster allocations】= 3、1、1、1，【remainders】= 3         // 9*2/5=3、9*1/5=1
+2、Sorting：member1 > member2 = member3 = member4
+3、One possible result：4、2、2、1
+
+Supposing sum replicas change from 9 to 8
+
+1、Calculation：【each cluster allocations】= 3、1、1、1，【remainders】= 2         // 8*2/5=3、8*1/5=1
+2、Sorting：member1 > member2 = member3 > member4                                // member2 has more current replicas than member3 / member4
```

Comment:
I don't get why `member2` has more current replicas than `member3` and `member4`.

Reply:
nice finding, mistake in writing, actually is member2 and member3 has more current replicas than member4